### PR TITLE
Route parameters sync

### PR DESCRIPTION
**Pull request by nnngrach**
[issue](https://github.com/osmandapp/OsmAnd-iOS/issues/3761)

Synced that code:
RouteParametersFragment.setupRoutingPrefs() https://github.com/osmandapp/Osmand/blob/995a48fef57d9c387687160fb1c05bc2bbc36830/OsmAnd/src/net/osmand/plus/settings/fragments/RouteParametersFragment.java#L207
RouteParametersFragment.setupOsmAndRouteServicePrefs() https://github.com/osmandapp/Osmand/blob/995a48fef57d9c387687160fb1c05bc2bbc36830/OsmAnd/src/net/osmand/plus/settings/fragments/RouteParametersFragment.java#L247

---

before (truck, region usa)
Screenshot 2024-06-17 at 18 24 37

after (truck, region usa)
Screenshot 2024-06-21 at 15 14 20
Screenshot 2024-06-21 at 15 13 24
Screenshot 2024-06-21 at 15 13 34

---
driving
Screenshot 2024-06-21 at 15 10 14

---
cycling
Screenshot 2024-06-21 at 15 12 16

---
walking
Screenshot 2024-06-21 at 15 16 03

---
train
Screenshot 2024-06-21 at 15 14 59

---
aircraft
Screenshot 2024-06-21 at 15 16 39

---
boat
Screenshot 2024-06-21 at 15 19 57

---
motorcycle
Screenshot 2024-06-21 at 15 17 16

---
moped
Screenshot 2024-06-21 at 15 21 11

---
riding
Screenshot 2024-06-21 at 15 21 57

---
skiing
Screenshot 2024-06-21 at 15 18 12 Screenshot 2024-06-21 at 15 18 23 Screenshot 2024-06-21 at 15 18 46

### CODE REVIEW

**Comment by nnngrach** [comment]
There is differing order in the first section because cpp parser return UnorderedMap.

Screenshot 2024-06-21 at 15 54 15
Screenshot 2024-06-21 at 15 53 42

File: Sources/Controllers/Settings/NavigationSettings/OARouteParametersViewController.mm [hunk 1]
```diff
@@ -115,284 +146,452 @@ - (BOOL) checkIfAnyParameterIsSelected:(vector <RoutingParameter>)routingParamet
...
+    }
+    [self setupTimeConditionalRoutingPref:tableSection];
+}
+
+- (void) setupHazmatUSACategoryPreference:(NSMutableArray *)tableSection
+{
+    if ([_settings.drivingRegion get:self.appMode] == DR_US)
+    {
+        NSArray<NSArray<NSString *> *> *fetchedParams = [self getHazmatUsaParamsIds];
+        NSMutableArray<NSString *> *paramsIds = fetchedParams[0];
```
Comment (tigrim) [inline Sources/Controllers/Settings/NavigationSettings/OARouteParametersViewController.mm:353]: add check fetchedParams.count > ..

File: Sources/Controllers/Settings/NavigationSettings/RouteParameterHazmatUsa.swift [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+//  RouteParameterHazmatUsa.swift
+//  OsmAnd Maps
+//
+//  Created by Max Kojin on 20/06/24.
+//  Copyright © 2024 OsmAnd. All rights reserved.
+//
+
+import UIKit
+
+@objc(OARouteParameterHazmatUsa)
```
Comment (tigrim) [inline Sources/Controllers/Settings/NavigationSettings/RouteParameterHazmatUsa.swift:11]: don't have to write

File: Sources/Controllers/Settings/NavigationSettings/RouteParameterHazmatUsa.swift [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        false
+    }
+    
+    override func generateData() {
+        tableData.clearAllData()
+        let section = tableData.createNewSection()
+        
+        for i in 0 ..< parameterIds.count {
+            let row = section.createNewRow()
+            row.cellType = OASwitchTableViewCell.getIdentifier()
```
Comment (tigrim) [inline Sources/Controllers/Settings/NavigationSettings/RouteParameterHazmatUsa.swift:61]: reuseIdentifier?

**Review by tigrim — APPROVED** [review]

**Pull request merged**